clang/include/clang/CIR/Dialect/IR/CIRTypesDetails.h

```diff
@@ -35,11 +35,14 @@ struct StructTypeStorage : public TypeStorage {
     bool packed;
     StructType::RecordKind kind;
     ASTRecordDeclInterface ast;
+    // Extra unique key to differentiate replacements on mutable parts
+    int replaceVersion;
```

Comment:
This struct change sounds super adhoc and specific, doesn't fit the overall picture. Please elaborate why we need this and what are you trying to do, so perhaps we can find a different solution?

Reply:
Sure. The whole `AttrTypeSubElementHandler` was taken from LLVMStructType, which is necessary for pure C++ types as such to support walk and replace infrastructure in MLIR -- It's necessary for the AS lowering pass: `recursivelyReplaceElementsIn` to work.
https://github.com/llvm/clangir/blob/80e1a10002fce76d0880ef15ff8f481dcad626c1/mlir/lib/Dialect/LLVMIR/IR/TypeDetail.h#L333-L355

But they left a TODO comment for the mutable case. I thought an extra tracing key would work, so gave it a try.